staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -187,6 +181,41 @@ func ValidateCustomResourceDefinitionSpec(spec *apiextensions.CustomResourceDefi
 		}
 	}
 
+	ValidateCustomResourceConversion(spec.Conversion, fldPath.Child("conversion"))
+
+	return allErrs
+}
+
+func validateEnumStrings(fldPath *field.Path, value string, accepted []string, required bool) field.ErrorList {
+	if required && value == "" {
+		return field.ErrorList{field.Required(fldPath, "")}
+	}
+	for _, a := range accepted {
+		if a == value {
+			return field.ErrorList{}
+		}
+	}
+	return field.ErrorList{field.NotSupported(fldPath, value, accepted)}
+}
+
+// ValidateCustomResourceConversion statically validates
+func ValidateCustomResourceConversion(conversion *apiextensions.CustomResourceConversion, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if conversion == nil {
+		return allErrs
+	}
+	allErrs = append(allErrs, validateEnumStrings(fldPath.Child("strategy"), string(conversion.Strategy), []string{string(apiextensions.NoneConverter), string(apiextensions.WebhookConverter)}, true)...)
+	if conversion.Strategy == apiextensions.WebhookConverter {
+		if conversion.WebhookClientConfig == nil {
+			if utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceWebhookConversion) {
+				allErrs = append(allErrs, field.Required(fldPath.Child("webhookClientConfig"), "required field when strategy is set to Webhook"))
+			} else {
+				allErrs = append(allErrs, field.Required(fldPath.Child("webhookClientConfig"), "required field when strategy is set to Webhook. The field may have been auto-cleared when CustomResourceWebhookConversion feature is disabled"))
```

Comment:
would "required when strategy is set to Webhook, but not allowed because the CustomResourceWebhookConversion feature is disabled" be clearer?

Reply:
well, that is not accurate. we don't know if user didn't provide them, or it was provided but auto-cleared.

Reply:
whether it's cleared or not is an internal detail... what's important to the user is that this field is required if strategy is set to Webhook, but cannot be set because the CustomResourceWebhookConversion feature is disabled